.github/actions/use-pre-commit/action.yml

```diff
@@ -45,6 +49,37 @@ runs:
         python ~/.cache/pre-commit/pre-commit.pyz --version
       shell: bash
 
+    - name: Retrieve the list of all changed files
+      uses: dorny/paths-filter@4512585405083f25c027a35db413c2b3b9006d50  # pin v2.11.1
+      if: ${{ inputs.check-all-files != 'true' }}
+      id: changes
+      with:
+        list-files: shell
+        filters: |
+          all:
+            - added|modified: '**'
+
+    - name: Build pre-commit args
+      id: pre-commit-args
+      run: |
+        # Build pre-commit args
+        if [ '${{ inputs.check-all-files }}' != 'true' ]; then
+          for file in $ALL_FILES; do
+            ARGS="--files=$file $ARGS"
+          done
+        else
+          ARGS="--all-files $ARGS"
+        fi
+        echo "args=$ARGS" >> $GITHUB_OUTPUT
+      env:
+        ALL_FILES: ${{ steps.changes.outputs.all_files }}
+        ARGS: "${{ inputs.extra-args }}"
+      shell: bash
+
     - name: Run pre-commit
-      run: python ~/.cache/pre-commit/pre-commit.pyz run --show-diff-on-failure --color=always ${{ inputs.extra-args }}
+      run: >-
+        python ~/.cache/pre-commit/pre-commit.pyz
```

Comment:
`>-` is hard to read (it seems like `run --show-diff-on-failure` is a separate command)
you should add indent to improve this:
```
python ~/.cache/pre-commit/pre-commit.pyz
    run --show-diff-on-failure
    --color=always
```

but even like that I'm not a great fan, I'd rather use a `|` with backslashes to stay within bash expectations:
```
run: |
  python ~/.cache/pre-commit/pre-commit.pyz \
    run --show-diff-on-failure \
    --color=always
```